examples/website/collision-filter/app.jsx

```diff
@@ -0,0 +1,137 @@
+import React, {useEffect, useMemo, useState} from 'react';
+import {createRoot} from 'react-dom/client';
+import {Map} from 'react-map-gl';
+import maplibregl from 'maplibre-gl';
+import {WebMercatorViewport} from '@deck.gl/core';
+import DeckGL from '@deck.gl/react';
+import {GeoJsonLayer, TextLayer} from '@deck.gl/layers';
+import {CollisionFilterExtension} from '@deck.gl/extensions';
+import * as turf from '@turf/turf';
+
+const DATA_URL =
+  'https://raw.githubusercontent.com/visgl/deck.gl-data/master/examples/collision-filter/ne_10_roads_mexico.json';
+const LINE_COLOR = [0, 173, 230];
+
+const initialViewState = {longitude: -100, latitude: 24, zoom: 5, minZoom: 5, maxZoom: 12};
+
+function calculateLabels(data, pointSpacing) {
+  const routes = data.features.filter(d => d.geometry.type !== 'Point');
+
+  // Add points along the lines
+  const filteredLabels = routes.map(d => {
+    const lineLength = Math.floor(turf.lineDistance(d.geometry));
+
+    const result = [];
```

Comment:
Move `result` declaration out of the loop and skip the `reduce` step in the end:

```js
const result = [];

const addPoint = ...

for (const feature of routes) {
  // generate points
}

return result;
```